2007-01-22  Ulrich Drepper  <drepper@redhat.com>

       [BZ #3902]
       * stdio-common/_itoa.c (_itoa): Make sure at least a zero is emitted.

===================================================================
RCS file: /cvs/glibc/libc/stdio-common/_itoa.c,v
retrieving revision 1.21
retrieving revision 1.22
diff -u -r1.21 -r1.22
--- libc/stdio-common/_itoa.c	2004/06/06 06:02:14	1.21
+++ libc/stdio-common/_itoa.c	2007/01/22 16:16:08	1.22
@@ -269,6 +269,7 @@
 
     default:
       {
+	char *bufend = buflim;
 #if BITS_PER_MP_LIMB == 64
 	mp_limb_t base_multiplier = brec->base_multiplier;
 	if (brec->flag)
@@ -454,6 +455,8 @@
 	  }
 	while (n != 0);
 #endif
+	if (buflim == bufend)
+	  *--buflim = '0';
       }
       break;
     }
